File: iteracion 2/Instruction.c
#include <stdio.h>

struct instruction{
    int op = 0;
    int r = 0;
    int n = 0;
    int d = 0;
};


struct instruction write(int op, int r, int n, int d){
    struct instruction inst;
    inst.op = op;
    inst.r = r
    inst.n = n;
    inst.d = d
    return inst;
}

//void read(){}

File: iteracion 3/js/instruction.js
class Instruction
{
    struct instruction 
    this.op = 0;
    this.r = 0;
    this.n = 0;
    this.d = 0;
    // Represents TAM instructions.
    op = 0;
    // OpCode
    r = 0;
    // RegisterNumber
    n = 0;
    // Length
    d = 0;
    // Operand
    write(output) throws IOException
    {
        output.writeInt(this.op);
        output.writeInt(this.r);
        output.writeInt(this.n);
        output.writeInt(this.d);
    }
    static read(input) throws IOException
    {
        var inst = new Instruction();
        try
        {
            inst.op = input.readInt();
            inst.r = input.readInt();
            inst.n = input.readInt();
            inst.d = input.readInt();
            return inst;
        } catch (var s = null)
        {
            return null;
        }
    }
}